finagle-base-http/src/main/scala/com/twitter/finagle/http/CookieMap.scala

```diff
@@ -103,13 +103,13 @@ class CookieMap private[finagle] (message: Message, cookieCodec: CookieCodec)
   def getValue(name: String): Option[String] = get(name).map(_.value)
 
   /**
-   * Fetches all cookies with the given `name` from this map.
+   * Fetches all cookies with the given `name` from this map, in order of insertion.
```

Comment:
suggestion:
from the RFC:
https://datatracker.ietf.org/doc/html/rfc6265#section-4.2.2
> servers SHOULD NOT rely upon the order in which these cookies appear in the header

we shouldn't make the ordering part of the contract, imo

Reply:
The way I see it:
- The RFC does also recommend that the client use a specific order in https://datatracker.ietf.org/doc/html/rfc6265#section-5.4, although that's only a "should", not a "must" (hence why it shouldn't be relied upon). Major browsers do respect that AFAIK.
- The recommendation not to rely on the order is, again only a "should not", not a "must not"; up to you to see if the HTTP library should be opinionated on that, I guess?
- I think it would be a bit surprising to have a `Seq` (rather than e.g. a `Set`) but with values in a potentially "random" order, different from the order we see when looking at the actual HTTP request.
- There may also be a question about the behavior of `get()`/`getValue()` (which use `headOption`) when several values are present. Always taking the first value in the list as sent by the client seems reasonable to me?

Reply:
> The recommendation not to rely on the order is, again only a "should not", not a "must not"; up to you to see if the HTTP library should be opinionated on that, I guess?

I don't think we need to be opinionated about it, and for that reason, I think it's nice that the implementation preserves order, but also that we don't put into the contract that it does.  Servers can still be well-behaved and ignore the order, and it's better for people to program as if that's the case.  If people want to program in a way that's explicitly discouraged by the spec they can for now, but support for them may be dropped in the future.

> There may also be a question about the behavior of get()/getValue() (which use headOption) when several values are present. Always taking the first value in the list as sent by the client seems reasonable to me?

I agree it's reasonable, but I don't think that we should make it part of the contract.

With that said, I don't feel super strongly about this.  It's just my opinion.  I'm OK with shipping it even if you decide not to change it!